app/cdap/components/PipelineConfigurations/ConfigurationsContent/ConfigModelessActionButtons/index.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  updatePipeline,
+  runPipeline,
+  schedulePipeline,
+  updatePreferences,
+} from 'components/PipelineConfigurations/Store/ActionCreator';
+import { setRunError } from 'components/PipelineDetails/store/ActionCreator';
+import ConfigModelessSaveBtn from 'components/PipelineConfigurations/ConfigurationsContent/ConfigModelessActionButtons/ConfigModelessSaveBtn';
+import { Observable } from 'rxjs/Observable';
+import PipelineConfigurationsStore from 'components/PipelineConfigurations/Store';
+import { convertKeyValuePairsToMap } from 'services/helpers';
+import { GLOBALS } from 'services/global-constants';
+
+require('./ConfigModelessActionButtons.scss');
+
+interface IConfigModelessActionButtonsProps {
+  onClose: () => void;
+  isHistoricalRun?: boolean;
+  applyRuntimeArguments?: (runtimeArgs: any) => void;
+  isDeployed?: boolean;
+  isPreview?: boolean;
+  studioRunPipeline?: () => void;
+  applyBatchConfig?: (...args) => void;
+  applyRealtimeConfig?: (...args) => void;
+  pipelineType?: string;
+}
+
+export default class ConfigModelessActionButtons extends Component<
+  IConfigModelessActionButtonsProps
+> {
+  public state = {
```

Comment:
For newly created component, I guess we're mostly using `useState` or `useReduce` for state control? And hence we prefer using functional component which support such hooks. @seanfeld Any input?

Reply:
This was changing from js to tsx. @seanfeld asked me to just change the interface and not using proptypes

Reply:
Ok, this one is just tsx conversion, actually not new code, I see

Reply:
i agree with both of you - theres only one exception to a new component being a class https://reactjs.org/docs/error-boundaries.html but other than that all new stuff needs to use hooks, but like sumeng said this is just a conversion :)